sklearn/cluster/tests/test_mean_shift.py

```diff
@@ -39,21 +39,23 @@ def test_estimate_bandwidth():
     assert 0.9 <= bandwidth <= 1.5
 
 
-def test_estimate_bandwidth_1sample():
+def test_estimate_bandwidth_1sample(global_dtype):
     # Test estimate_bandwidth when n_samples=1 and quantile<1, so that
     # n_neighbors is set to 1.
-    bandwidth = estimate_bandwidth(X, n_samples=1, quantile=0.3)
+    bandwidth = estimate_bandwidth(X.astype(global_dtype), n_samples=1, quantile=0.3)
```

Comment:
```suggestion
    bandwidth = estimate_bandwidth(X.astype(global_dtype, copy=False), n_samples=1, quantile=0.3)
```
and everywhere you use astype. We don't want the unnecessary copy when X already has the requested type